src/cudamatrix/cu-device.h

```diff
@@ -55,7 +55,12 @@ class CuDevice {
   inline void* Malloc(size_t size) { return allocator_.Malloc(size); }
 
   inline void* MallocPitch(size_t row_bytes, size_t num_rows, size_t *pitch) {
-    return allocator_.MallocPitch(row_bytes, num_rows, pitch);
+    if (random_stride_mode_) {
+      return allocator_.MallocPitch(row_bytes + 128 * (rand() & 1), num_rows,
```

Comment:
doesn't matter, as it will be rounded up to a multiple of 128 bytes anyway
to get the stride.  adding 128 puts it in a different bucket.

On Tue, Aug 23, 2016 at 7:31 PM, Daniel Galvez <notifications@github.com>
wrote:

> In src/cudamatrix/cu-device.h
> <https://github.com/kaldi-asr/kaldi/pull/997#discussion_r75971055>:
>
> > @@ -55,7 +55,12 @@ class CuDevice {
> >    inline void* Malloc(size_t size) { return allocator_.Malloc(size); }
> >
> >    inline void* MallocPitch(size_t row_bytes, size_t num_rows, size_t *pitch) {
> > -    return allocator_.MallocPitch(row_bytes, num_rows, pitch);
> > +    if (random_stride_mode_) {
> > +      return allocator_.MallocPitch(row_bytes + 128 * (rand() & 1), num_rows,
>
> I actually question the usage of 128, as CUDA allocations tend to be
> aligned on some multiple of 2 (like 64) anyway. I would exepct that 129
> would be more likely to expose buggy code.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/997/files/ccd2b85ea256cfe1e3f6ad76c04c4e260a0e6344#r75971055>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu1afHQFqcRs38amOOonERkgNof-Gks5qi4LqgaJpZM4Jqu95>
> .
>